src/app/util/attribute-storage.cpp

```diff
@@ -878,6 +878,23 @@ static uint16_t findIndexFromEndpoint(EndpointId endpoint, bool ignoreDisabledEn
     return kEmberInvalidEndpointIndex;
 }
 
+uint16_t emberAfGetClusterServerEndpointIndex(EndpointId endpoint, ClusterId cluster)
```

Comment:
Given the behavior of this function, the way on-off-server is using it is wrong and will lead to out-of-bounds access on the array.

Reply:
Again, what I think we want here, given how the array in on-off-server is set up, is a function that:

1) For static endpoins returns what emberAfFindClusterServerEndpointIndex returns.
2) For dynamic endpoints returns the index of the endpoint minus FIXED_ENDPOINT_COUNT plus the number of fixed endpoints that implement the given cluster.

or something like that.  Please check carefully.

This code _really_ needs tests.

Reply:
Argh I confused `EMBER_AF_ON_OFF_CLUSTER_SERVER_ENDPOINT_COUNT`  with `FIXED_ENPOINT_COUNT` to be the same.   :Facepalm: 

Alright I'll fix that tomorrow.

For a unit test, I agree but I don't have a simple way to do it without re-implemeting the app-util mock, and I don't currently have the time to do that part.

I am willing to do with but it will have to wait a bit

Reply:
Yeah, I'm not going to block this PR on the test bit; just saying that it would be really good to have them in general....

Reply:
Updated the function to act as you described, Sorry for my confusion and mistake yesterday. 

I'll now plan time to unit test that code in the incoming days